torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -5635,3 +5660,81 @@ def test_rref_to_here_synchronization_cross_device(self):
                 self.assertEqual(actual, expected)
 
         rpc.shutdown()
+
```

Comment:
Shall we also add a test for the following error message?

```c++
    TORCH_CHECK_VALUE(
        excessDevices.empty(),
        "The result contained tensors residing on devices ",
        formatSetOfDevices(excessDevices),
        " which are not among the expected devices ",
        formatSetOfDevices(*devices_));
```

Reply:
Done, and in fact that caught a bug (i.e., `DeviceIndex`es are `int8_t`s, a.k.a., chars, hence outputting them to a stringstream won't format them as numbers).